src/plugins/profiling/server/routes/flamegraph.ts

```diff
@@ -77,20 +36,68 @@ export class FlameGraph {
   stackframes: Map<StackFrameID, StackFrame>;
   executables: Map<FileID, Executable>;
 
+  private readonly logger: Logger;
+
   constructor(
     sampleRate: number,
     totalCount: number,
     events: Map<StackTraceID, number>,
     stackTraces: Map<StackTraceID, StackTrace>,
     stackFrames: Map<StackFrameID, StackFrame>,
-    executables: Map<FileID, Executable>
+    executables: Map<FileID, Executable>,
+    logger: Logger
   ) {
     this.sampleRate = sampleRate;
     this.totalCount = totalCount;
     this.events = events;
     this.stacktraces = stackTraces;
     this.stackframes = stackFrames;
     this.executables = executables;
+    this.logger = logger;
+  }
+
+  private getExeFileName(exe: any, type: number) {
```

Comment:
we later need this function and getLabel() for the callgraph as well. So function outside the class seem to be better suited.